File: src/main/java/org/wcardinal/controller/internal/info/RejectionInfo.java
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

package org.wcardinal.controller.internal.info;

import java.util.Map;

public class RejectionInfo extends Info<Boolean, RejectionInfo> {
	public RejectionInfo( final Map<String, Boolean> nameToData, final Map<String, RejectionInfo> nameToInfo ) {
		super( nameToData, nameToInfo );
	}

	public static RejectionInfo create( final Map<String, Boolean> nameToData, final Map<String, RejectionInfo> nameToInfo ) {
		if( nameToData == null && nameToInfo == null ) return null;
		return new RejectionInfo( nameToData, nameToInfo );
	}
}

File: src/main/typescript/wcardinal/controller/internal/info/rejection-info.ts
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

import { RejectionInfoDataMap } from "./rejection-info-data-map";
import { RejectionInfoMap } from "./rejection-info-map";

export interface RejectionInfo {
	[ 0 ]: RejectionInfoDataMap | null;
	[ 1 ]: RejectionInfoMap | null;
}
